course/src/isometry.cpp

```diff
@@ -10,9 +10,146 @@
 namespace ekumen {
 namespace math {
 
-/*
- * your code here
- */
+namespace {
+
+Vector3 translation_vector_operation(const Matrix3& matrix,
+                                     const Vector3& first_vector,
+                                     const Vector3& second_vector) {
+  Vector3 aux;
+  aux = matrix * first_vector + second_vector;
+  return aux;
+}
+
+}  // namespace
+
+
+Isometry Isometry::fromTranslation(const Vector3& r_vector) {
+  Isometry aux;
+  aux.translation_vector = r_vector;
+  return aux;
+}
+
+Vector3 Isometry::translation() const {
+  Vector3 aux{translation_vector};
+  return aux;
+}
+
+Vector3 Isometry::transform(const Vector3& r_vector) const {
+  Vector3 aux;
+  aux = (*this) * r_vector;
+  return aux;
+}
+
+Isometry Isometry::inverse() const {
+  double deter = rotation_matrix.det();
+  if (deter == 0) {
+    throw std::invalid_argument("Isometry doesn't have an inverse");
+  }
+  Matrix3 aux;
+  aux = aux.transpose();
```

Comment:
What's the transpose of the following matrix?
```
1 0 0
0 1 0
0 0 1
```

Reply:
It's the same, right?

Reply:
Oh I get it, that first matrix should be initialized with the values of the class matrix, not with default values.

Reply:
Yes. However my comment should have been regarding a zero matrix.
Check that you created a zero matrix and then you got the transpose, which is the same.

Reply:
Yeah, that zero matrix was not intended, the idea was to transpose the matrix of `this` and then begin to operate, I'll correct.

Reply:
Addressed.